qemu/tests/virtio_mem_numa_basic.py

```diff
@@ -0,0 +1,69 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_misc import NumaInfo
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Memory share and discard-data hotplug test
+    1) Boot guest with two virtio-mem devices
+    2) Check virtio-mem devices
+    3) Resize virtio-mem devices
+    4) Check virtio-mem devices
+    5) Resize virtio-mem devices to the maximum
+    6) Check virtio-mem devices
+    7) Resize virtio-mem devices to the minimum
+    8) Check virtio-mem devices
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    host_numa_node = NumaInfo()
+    node_list = host_numa_node.online_nodes_withmem
+    if len(node_list) < 2:
+        test.cancel("This host only has one NUMA node, skipping test...")
+
+    timeout = int(params.get("login_timeout", 240))
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+
+    virtio_mem_utils.check_memory_devices("requested-size_memory_vmem", params, threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem("requested-size_memory_vmem", params, threshold, session, vm, test)
+
+    requested_size_vmem0 = int(float(normalize_data_size(params.get("requested-size_memory-vmem0"), 'B')))
+    vm.monitor.qom_set("virtio_mem-vmem0", "requested-size", requested_size_vmem0)
+    requested_size_vmem1 = int(float(normalize_data_size(params.get("requested-size_memory-vmem1"), 'B')))
+    vm.monitor.qom_set("virtio_mem-vmem1", "requested-size", requested_size_vmem1)
+    time.sleep(10)
+
+    virtio_mem_utils.check_memory_devices("requested-size_memory-vmem", params, threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem("requested-size_memory-vmem", params, threshold, session, vm, test)
+
+    requested_size_max_vmem0 = int(float(normalize_data_size(params.get("requested-size_max_memory-vmem0"), 'B')))
+    vm.monitor.qom_set("virtio_mem-vmem0", "requested-size", requested_size_max_vmem0)
+    requested_size_max_vmem1 = int(float(normalize_data_size(params.get("requested-size_max_memory-vmem1"), 'B')))
+    vm.monitor.qom_set("virtio_mem-vmem1", "requested-size", requested_size_max_vmem1)
+    time.sleep(30)
+
+    virtio_mem_utils.check_memory_devices("requested-size_max_memory-vmem", params, threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem("requested-size_max_memory-vmem", params, threshold, session, vm, test)
+
+    vm.monitor.qom_set("virtio_mem-vmem0", "requested-size", params.get_numeric("requested-size_min_memory-vmem0"))
+    vm.monitor.qom_set("virtio_mem-vmem1", "requested-size", params.get_numeric("requested-size_min_memory-vmem1"))
+    time.sleep(180)
```

Comment:
The current release time is too long because of a product bug. Under normal circumstances, it should be completed within the 30s, so I think it can be changed to 30s.

Reply:
Changed, thanks!